hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -468,18 +470,21 @@ private void uploadBlockAsync(AbfsBlock blockToUpload,
               try {
                 LOG.debug("Append over Blob for ingress config value {} for path {} ",
                         client.getAbfsConfiguration().shouldIngressFallbackToDfs(), path);
+                TracingContext tracingContextBlobAppend = new TracingContext(tracingContext);
+                tracingContextBlobAppend.setFallbackDFSAppend("B " + offset);
                 op = client.append(blockToUpload.getBlockId(), path, blockUploadData.toByteArray(), reqParams,
-                        cachedSasToken.get(), new TracingContext(tracingContext), getETag());
+                        cachedSasToken.get(), tracingContextBlobAppend, getETag());
                 String key = blockToUpload.getBlockId();
-                if (!getMap().containsKey(key)) {
-                  throw new Exception("Block is missing with blockId " + blockToUpload.getBlockId());
-                } else {
+                try {
                   mapLock.lock();
-                  try {
+                  if (!getMap().containsKey(key)) {
+                    throw new Exception("Block is missing with blockId " + blockToUpload.getBlockId() +
```

Comment:
Should we add streamId as well into the log ?

Reply:
Added